src/app/core/services/show-style-variant-state.service.ts

```diff
@@ -3,13 +3,40 @@ import { BehaviorSubject, lastValueFrom, Observable } from 'rxjs'
 import { ShowStyleVariant } from '../models/show-style-variant'
 import { ShowStyleVariantService } from '../abstractions/show-style-variant.service'
 import { EventSubscription } from '../../event-system/abstractions/event-observer.service'
+import { RundownEventObserver } from './rundown-event-observer.service'
+import { RundownUpdatedEvent } from '../models/rundown-event'
+import {Logger} from "../abstractions/logger.service";
 
 @Injectable()
 export class ShowStyleVariantStateService implements OnDestroy {
   private readonly showStyleVariantSubjects: Map<string, BehaviorSubject<ShowStyleVariant>> = new Map()
   private readonly eventSubscriptions: EventSubscription[]
+  private readonly logger: Logger
 
-  constructor(private readonly showStyleVariantService: ShowStyleVariantService) {}
+  constructor(
+    private readonly showStyleVariantService: ShowStyleVariantService,
+    private readonly rundownEventObserver: RundownEventObserver,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('ShowStyleVariantStateService')
+    this.subscribeToRundownEvents()
+  }
+
+  private subscribeToRundownEvents(): void {
+    this.rundownEventObserver.subscribeToRundownUpdates(this.updateShowStyleVariantFromEvent.bind(this))
+  }
+
+  private updateShowStyleVariantFromEvent(event: RundownUpdatedEvent): void {
```

Comment:
We should really stop using the `fromEvent` pattern that has spread throughout the frontend...
 
The "from event" references something outside the scope of this method and the only thing that you actually need from `RundownUpdatedEvent` is the `rundownId`. And since you have "bound" the method declaration to something outside of the scope of the method, you limit the flexibility in where it could be used. Right now, it can _only_ be used when you listen for RundownEvents, which is quite inflexible and goes against the OpenClose principle.

By making the signature something like `updateShowStyleVariantForRundown(rundownId: string)` you have opened this method up to be used by a plethora of places and you can now extend your program to use this method somewhere else, without having to first modify it (the basis of the OpenClose principle).

Reply:
I see you point, and i have updated the method to your suggestion (even your naming though that was accidental).